File: nodejs/src/application/use-cases/test-post.use-case.ts
import { Injectable } from '@nestjs/common';
import {NubosaApiService} from "../../infrastructure/external/nubosa-api.service";
import {UpsertCompanyDto} from "../dto/upsert-company.dto";

// Simple use case for demonstration purposes
@Injectable()
export class TestPostUseCase {
  constructor(private nubosaService: NubosaApiService) {
  }

  execute() {
    return this.nubosaService.post({
      certifiedPass: 'certifiedPass',
      name: 'Example company',
      document: '11111111',
      commercialName: 'Excom',
    } as UpsertCompanyDto);
  }
}

File: java/src/main/java/com/example/application/use_cases/TestPostUseCase.java
package com.example.application.use_cases;

import com.example.application.dto.ResponseDto;
import com.example.application.dto.UpsertCompanyDto;
import com.example.infraestructure.external.NubosaApiService;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.io.IOException;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;

@Service
@AllArgsConstructor
public class TestPostUseCase {
    private final NubosaApiService nubosaApiService;

    public ResponseDto execute() throws NoSuchAlgorithmException, IOException, InvalidKeyException, InterruptedException {
        return this.nubosaApiService.post(new UpsertCompanyDto(
                "11111111",
                "Example company",
                "Excom",
                "certifiedPass"
        ));
    }
}
